controllers/nodenetworkconfigurationpolicy_controller.go

```diff
@@ -160,16 +159,12 @@ func (r *NodeNetworkConfigurationPolicyReconciler) initializeEnactment(policy nm
 	})
 }
 
-func (r *NodeNetworkConfigurationPolicyReconciler) maxUnavailableNodeCount(policy *nmstatev1beta1.NodeNetworkConfigurationPolicy) (int, error) {
-	nmstateNodes, err := node.NodesRunningNmstate(r.Client)
-	if err != nil {
-		return 0, err
-	}
+func (r *NodeNetworkConfigurationPolicyReconciler) maxUnavailableNodeCount(policy *nmstatev1beta1.NodeNetworkConfigurationPolicy, matchingNodes int) (int, error) {
```

Comment:
Remove the `matchingNodes` argument and call `enactmentCount.Matching()` inside at the end this is what we want to do, we don't need to parameterize it also remove the `(r *NodeNetworkConfigurationPolicyReconciler)` receiver since is not used, we can even move this undier ./pkg and make it public to use it at integratin tests.